test/testdrive/protobuf-import.td

```diff
@@ -0,0 +1,105 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Test that Protobuf imports are handled correctly.
+
+$ set importee-schema
+syntax = "proto3";
+
+message Importee1 {
+    bool b = 1;
+}
+
+message Importee2 {
+    int32 i = 1;
+}
+
+
+$ set importer-schema
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
+import "importee.proto";
+
+message Importer {
+    Importee1 importee1 = 1;
+    Importee2 importee2 = 2;
+    google.protobuf.Timestamp ts = 3;
+}
+
+
+# First, test without the schema registry.
+
+$ file-append path=importee.proto
+\${importee-schema}
+
+$ file-append path=importer.proto
+\${importer-schema}
+
+$ protobuf-compile-descriptors inputs=importee.proto,importer.proto output=import.pb
+
+$ kafka-create-topic topic=import partitions=1
+
+$ kafka-ingest topic=import format=protobuf descriptor-file=import.pb message=.Importer
+{"importee1": {"b": false}, "importee2": {"i": 42}, "ts": {"seconds": 1234, "nanos": 5678}}
+
+> CREATE MATERIALIZED SOURCE import FROM
+  KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-import-${testdrive.seed}'
+  FORMAT PROTOBUF MESSAGE '.Importer' USING SCHEMA FILE '${testdrive.temp-dir}/import.pb'
+
+> SELECT importee1::text, importee2::text, ts::text, mz_offset FROM import
+importee1  importee2  ts            mz_offset
+---------------------------------------------
+(f)        (42)       (1234,5678)   1
+
+# Then, test again with the Confluent Schema Registry. Publishing Protobuf
+# schemas to the CSR is tricky, because each Protobuf file needs to go into its
+# own subject. This is handled automatically by the Java Kafka client, but it is
+# complicated to duplicate that automatic logic in Testdrive. So we publish
+# the schemas manually here, but keep the schema very simple to reduce the pain.
+
+$ kafka-create-topic topic=import-csr partitions=1
+
+# The Confluent toolchain publishes even schemas for well-known types, so we
+# have to do the same.
+# See: https://github.com/protocolbuffers/protobuf/blob/61e0395c89fe520ae7569aea6838313195e05ec5/src/google/protobuf/timestamp.proto
+$ schema-registry-publish subject=google/protobuf/timestamp.proto schema-type=protobuf
+syntax = "proto3";
+
+package google.protobuf;
+
+message Timestamp {
+  int64 seconds = 1;
+  int32 nanos = 2;
+}
+
+$ schema-registry-publish subject=importee.proto schema-type=protobuf
+\${importee-schema}
+
+$ schema-registry-publish subject=testdrive-import-csr-${testdrive.seed}-value schema-type=protobuf references=importee.proto,google/protobuf/timestamp.proto
+\${importer-schema}
+
+# TODO(benesch): There is a big bug here! In the Confluent toolchain, these
```

Comment:
Can you link a task? since this is a pretty big one we should fix sooner rather than later!

Reply:
Yep, done!